src/Sulu/Bundle/TestBundle/Testing/AssertHttpStatusCodeTrait.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TestBundle\Testing;
+
+use Symfony\Component\HttpFoundation\Response;
+
+trait AssertHttpStatusCodeTrait
+{
+    /**
+     * Assert the HTTP status code of a Response.
+     *
+     * If the response is not as expected we set the assertion message to the
+     * body of the response - if it is json-decodable then we pretty print
+     * JSON.
+     *
+     * The $debugLength argument limits the number of lines included from the
+     * response body in case of failure.
+     */
+    protected static function assertHttpStatusCode(int $code, Response $response, int $debugLength = 50): void
+    {
+        $httpCode = $response->getStatusCode();
+
+        $message = '';
+        if ($code !== $httpCode) {
+            $message = $response->getContent();
+
+            if ('null' !== ($json = json_encode(json_decode($message, true), JSON_PRETTY_PRINT))
```

Comment:
Are you sure that's doing the same as the one before? Don't you want to move it over the exact same way it was before?

Reply:
Its an improved version we used in projects which handles also HTML correct with the debug length.